frontend/coprs_frontend/coprs/oidc.py

```diff
@@ -48,6 +48,28 @@ def oidc_enabled(config):
     )
 
 
+def oidc_username_from_userinfo(config, userinfo):
+    """
+    Return a unique user name from UserInfo
+    """
+    # Assume Ipsilon-style UserInfo (i.e., has a non-standard username
+    # claim that is guaranteed unique)
+    username = userinfo.get("username")
+    if username is None:
```

Comment:
This is going to work, but I think it's better to not guess here ... what if the `username` suddenly appears in the output in some of the OIDC providers?

I think that it would be better to just:

```python
unique_user_key = config.get("OIDC_USERNAMES", "preferred_username")
try:
  if unique_user_key == "preferred_username":
      username = userinfo["preferred_username"]
  elif unique_user_key == "username":
      username = userinfo["username"]
except KeyError:
  raise RuntimeError("Can't get unique username, check OIDC_USERNAMES configuration docs")
```

Reply:
Yeah, that makes sense. I'll do that.